### ISSUE TITLE
Naga AI broken

### ISSUE THREAD

**Issue by Matti-Koopa** [TeamTwilight/twilightforest#93]
The Naga is completely stuck. Does nothing, can't be hurt with any weapon.
I can push him and I do take damage.

This is a world that is 2 days old by now. I passed the Naga area a couple of times but never fought him.
![2017-06-22_15 12 07](https://user-images.githubusercontent.com/10476794/27435806-8608683e-575d-11e7-8d0d-259cc46a25bb.png)

Save file (I'm standing right by him) here: https://mega.nz/#!VBJGSIwa!Q95wOG7jD7fzy31EXMTGC2bz7H0ghbtVSv6u1kV57LM

TW Forest Build 42.

**Comment by Benimatic** [comment]
Have you experienced this problem in later builds?

**Comment by LordMonoxide** [comment]
I'm experiencing this on the latest commit to branch 1.11.2

**Comment by Tamaized** [comment]
Is there a way to reliably cause this? I've yet to see it happen myself.

**Comment by LapisWhovian** [comment]
1.**11**, Tama.

**Comment by LapisWhovian** [comment]
Bugs for unsupported versions are silly

**Comment by Tamaized** [comment]
I'm aware it's 1.11 however it's the same code, nothing has changed for the Naga between versions. I wasn't able to reproduce this back then either so if there's a way to reliably cause it then I'd like to know.

**Comment by williewillus** [comment]
is it because in that screenshot he doesn't have an arena/home position?

**Comment by LapisWhovian** [comment]
He enclosed the world with the issue, find out.

**Comment by Tamaized** [comment]
Not having a home is actually very possible to be the cause. Seems we even have some weird check for the Naga to not take damage if it's out of range of its home.

**Comment by Andromander** [comment]
Just spawned a Naga in an enclosed environment (a room dug out underground), and it doesn't move. However, above ground, it moves around as normal.

**Comment by LordMonoxide** [comment]
The naga had escaped its enclosure when it froze for me

**Comment by Tamaized** [comment]
Does the Naga only freeze when you're a certain distance away from it or when it freezes for you guys is it frozen even when you're up close to it?

**Comment by LordMonoxide** [comment]
It remained frozen and immune. I had to use `/kill` to kill it.

**Comment by LordMonoxide** [comment]
Haven't experienced this since moving to 1.12

**Comment by Marfjeh** [comment]
It did happen for me in 1.11.2 also i remembered there was an error thrown to the console. i cannot find the log at this moment but maybe when I'm home i can find the log file. 
also when i updated to 1.12 i wanted to do a Naga boss fight last time i could not attack it (it did move and attacked me). but that's maybe i didn't unlock the arena perhaps.

**Comment by Terrakota-d** [comment]
It happened to me too. When Naga went beyond the arena it stopped receiving damage but kept moving.  After I was ableto return Naga back it is back to normal.
Version 3.1.1, 136

**Comment by Tamaized** [comment]
Yea the Naga is meant to not take damage when its outside of its home range. I'm not quite sure if this is really needed or not.

**Comment by LordMonoxide** [comment]
Seems to me that if it's capable of breaking out on its own, it shouldn't become invincible

**Comment by GizmoTheMoonPig** [comment]
I think I can finally close this old-ass issue. 
I had to toy with the naga a bit to fix an issue opened recently, and I have deemed the naga as fully functional, even when outside of its home radius. Its AI will work fine when it leaves, and it will try to actively pathfind its way home instead of just sitting there (like it used to apparently?) or continuing to fight the player that can't deal damage to it. 
As for the naga not being able to take damage outside the courtyard... well, its code that beni himself wrote, and I'm not willing to change it. You shouldn't ever encounter the issue of not being able to hurt it now because it will always find its way back home.
